src/bucket/BucketManagerImpl.cpp

```diff
@@ -825,7 +825,183 @@ BucketManagerImpl::addBatch(Application& app, uint32_t currLedger,
                           liveEntries, deadEntries);
 }
 
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+
+void
+BucketManagerImpl::updateBucketListSizeAverage() const
+{
+    uint64_t sizeSum = 0;
+    for (auto const& size : mBucketListSizeSnapshots)
+    {
+        sizeSum += size;
+    }
+
+    mAverageBucketListSize = sizeSum / mBucketListSizeSnapshots.size();
+}
+
+void
+BucketManagerImpl::loadBucketListSizeWindow(AbstractLedgerTxn& ltxRoot) const
+{
+    LedgerTxn ltx(ltxRoot, false, TransactionMode::READ_ONLY_WITHOUT_SQL_TXN);
+    LedgerKey key(CONFIG_SETTING);
+    key.configSetting().configSettingID =
+        ConfigSettingID::CONFIG_SETTING_BUCKETLIST_SIZE_WINDOW;
+    auto txle = ltx.loadWithoutRecord(key, /*loadExpiredEntry=*/false);
+    releaseAssert(txle);
+    auto const& leVector =
+        txle.current().data.configSetting().bucketListSizeWindow();
+    mBucketListSizeSnapshots.clear();
+    for (auto e : leVector)
+    {
+        mBucketListSizeSnapshots.push_back(e);
+    }
+
+    updateBucketListSizeAverage();
+}
+
+void
+BucketManagerImpl::writeBucketListSizeWindow(AbstractLedgerTxn& ltxRoot) const
+{
+    releaseAssert(mAverageBucketListSize);
+    LedgerTxn ltx(ltxRoot);
+    LedgerKey key(CONFIG_SETTING);
+    key.configSetting().configSettingID =
+        ConfigSettingID::CONFIG_SETTING_BUCKETLIST_SIZE_WINDOW;
+    auto txle = ltx.load(key);
+    releaseAssert(txle);
+    auto& leVector = txle.current().data.configSetting().bucketListSizeWindow();
+    leVector.clear();
+    for (auto e : mBucketListSizeSnapshots)
+    {
+        leVector.push_back(e);
+    }
+
+    ltx.commit();
```

Comment:
We don't emit any meta here, do we? I think that's an issue.

Reply:
I don't think so? This is writing the sliding window to disk, which is a `CONFIG_ENTRY`. This is a Validator, BucketList specific entry that's only used to maintain the BucketList size average. The only BucketList size related meta we need to emit is a single value for the average. I think it's fine we don't emit anything hear because even though the sliding window is technically a `LedgerEntry`, it shouldn't be useful for any downstream systems and isn't a result of a transaction, so I don't think there would be a good place to put it in the meta and I don't think it would be useful.

Reply:
Sounds good, I somehow haven't noticed that we emit the average (and not the current) size.